rest-service/manager_rest/security/authentication.py

```diff
@@ -147,6 +147,16 @@ def _http_auth(self, user, username, password):
             )
         return user
 
+    @staticmethod
+    def _verify_token(token, user):
+        if not current_app.token_verified_cache.get_verify_hash_result(user, token):
+            if not verify_hash(compare_data=user.password, hashed_data=token):
+                raise_unauthorized_user_error(
+                    'Authentication failed for {0}'.format(user)
+                )
+            else:
+                current_app.token_verified_cache.cache_verify_hash_result(token, user)
```

Comment:
no reason for this to go back to the app, make the cache an instance attribute on this, and drop `@staticmethod`, imo better